include/HTTPRequest.hpp

```diff
@@ -58,12 +58,14 @@ class HTTPRequest final
      * @param onSuccess Callback to be called in case of success.
      * @param onError Callback to be called in case of error.
      * @param fileName File name of output file.
+     * @param retryAttempts Number of retry attempts if the query fails.
      */
     void get(
         const URL& url,
         std::function<void(const std::string&)> onSuccess,
         std::function<void(const std::string&)> onError = [](auto) {},
-        const std::string& fileName = "");
+        const std::string& fileName = "",
+        const unsigned int retryAttempts = 1);
```

Comment:
no retry as default value.

```suggestion
        const unsigned int retryAttempts = 0);
```